src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py

```diff
@@ -0,0 +1,219 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Centipede engine interface."""
+
+import os
+import re
+
+from clusterfuzz._internal.bot.fuzzers import dictionary_manager
+from clusterfuzz._internal.bot.fuzzers import engine_common
+from clusterfuzz._internal.bot.fuzzers import utils as fuzzer_utils
+from clusterfuzz._internal.system import environment
+from clusterfuzz._internal.system import new_process
+from clusterfuzz.fuzz import engine
+
+_CLEAN_EXIT_SECS = 10
+_TIMEOUT = 1200
+_SERVER_COUNT = 1
+_RSS_LIMIT = 4096
+_RLIMIT_AS = 5120
+_ADDRESS_SPACE_LIMIT = 0
+_DEFAULT_ARGUMENTS = [
+    '--exit_on_crash=1',
+    f'--timeout={_TIMEOUT}',
+    f'--fork_server={_SERVER_COUNT}',
+    f'--rss_limit_mb={_RSS_LIMIT}',
+    f'--address_space_limit_mb={_ADDRESS_SPACE_LIMIT}',
+]
+
+_CRASH_REGEX = re.compile('Crash detected, saving input to (.*)')
+
+
+class CentipedeError(Exception):
+  """Base exception class."""
+
+
+def _get_runner():
+  """Get the Centipede runner."""
+  centipede_path = os.path.join(environment.get_value('BUILD_DIR'), 'centipede')
```

Comment:
THis is formatted weirdly can you put os.path.join on the same line as the equal (49) and the rest on the next line.

Reply:
Could you please elaborate more on why it's weird?
I tried that but then found our auto-formatter changed the line back, maybe because the line is exactly 80 chars long.